mysite/views.py

```diff
@@ -1,3 +1,175 @@
+
+
+def profile_view(request):
+    user = request.user
+    return render(request, 'mysite/profile.html', {"user": user})
+
+
+class EditUserProfileView(UpdateView):
+    model = User
+    form_class = UserProfileForm
+    template_name = 'mysite/edit_profile.html'
+
+    def get_object(self, *args, **kwargs):
+        user = get_object_or_404(User, pk=self.kwargs['pk'])
+
+        # We can also get user object using self.request.user  but that doesnt work
+        # for other models.
+
+        return user
+
+    def get_success_url(self, *args, **kwargs):
+        return reverse("home")
+
+
+def change_password(request):
+
+    if request.method == 'POST':
+        form = PasswordChangeForm(request.user, request.POST)
+        if form.is_valid():
+            user = form.save()
+            update_session_auth_hash(request, user)  # Important!
+            messages.success(request, 'Your password was successfully updated!')
+            return redirect('profile')
+        else:
+            messages.error(request, 'Please correct the error below.')
+    else:
+        form = PasswordChangeForm(request.user)
+    return render(request, 'mysite/change_password.html', {
+        'form': form
+    })
+
+
+def project_list(request, template_name='mysite/project_list.html'):
+    projects = Project.objects.all()
+    data = {'object_list': projects}
+    return render(request, template_name, data)
+
+
+def project_create(request, template_name='mysite/project_form.html'):
+    form = ProjectForm(request.POST or None)
+    if form.is_valid():
+        project = form.save(commit=False)
+        project.created_by = request.user
+        project.status = 'Unassigned'
+        form.save()
+        return redirect('project_list')
+    return render(request, template_name, {'form': form})
+
+
+def project_view(request, pk, template_name='mysite/project_view.html'):
+    project = get_object_or_404(Project, pk=pk)
+    return render(request, template_name, {'project': project})
+
+
+def project_update(request, pk, template_name='mysite/project_form.html'):
+    project = get_object_or_404(Project, pk=pk)
+    form = ProjectForm(request.POST or None, instance=project)
+    if form.is_valid():
+        form.save()
+        return redirect('project_list')
+    return render(request, template_name, {'form': form})
+
+
+def project_delete(request, pk, template_name='mysite/project_confirm_delete.html'):
+    project = get_object_or_404(Project, pk=pk)
+    if request.method == 'POST':
+        project.delete()
+        return redirect('project_list')
+    return render(request, template_name, {'object': project})
+
+
+def project_bid_list(request, template_name='mysite/project_bid_list.html'):
+    project_bids = ProjectBid.objects.all()
+    data = {'object_list': project_bids}
+    return render(request, template_name, data)
+
+
+def project_bid_create(request, project_id, template_name='mysite/project_bid_form.html'):
```

Comment:
name should be more descriptive like "bin_for_project"

Reply:
Feedback Incorporated.